### PR TITLE
Added save command for adding hosts spawned elsewhere

### PR DESCRIPTION
**Pull request by olehermanse**
No support for private IPs (yet).

Looks like this:

```
$ cf-remote save -H root@198.211.105.11,root@128.199.44.119 --role hub --name hubs
$ cf-remote info -H hubs

root@198.211.105.11
OS            : ubuntu (debian)
Architecture  : x86_64
CFEngine      : 3.18.1 (Enterprise)
Policy server : 198.211.105.11
Binaries      : dpkg, apt

root@128.199.44.119
OS            : ubuntu (debian)
Architecture  : x86_64
CFEngine      : 3.18.1 (Enterprise)
Policy server : 128.199.44.119
Binaries      : dpkg, apt

$ cf-remote destroy --all
Destroying all hosts
Deleting saved group '@hubs' without terminating VMs:
  hubs-1: root@198.211.105.11 (hub)
  hubs-2: root@128.199.44.119 (hub)
```

### CODE REVIEW

File: cf_remote/commands.py [hunk 1]
```diff
@@ -358,6 +358,20 @@ def spawn(platform, count, role, group_name, provider=Providers.AWS, region=None
 
     return 0
 
+def _is_saved_group(vms_info, group_name):
+    group = vms_info[group_name]
+    return ("meta" not in group
+            or "region" not in group["meta"]
+            or "provider" not in group["meta"])
```
Comment (vpodzime) [inline cf_remote/commands.py:365]: I would actually add an explicit `"saved": true` to `meta` for such groups.

File: cf_remote/commands.py [hunk 2]
```diff
@@ -485,6 +509,31 @@ def init_cloud_config():
...
+    for index, host in enumerate(hosts):
+        split = host.split("@")
+        if len(split) != 2:
+            print("Host '{}' not accepted, must be given as user@ip-address".format(host))
+            return 1
+        user, ip = host.split("@")
+        instance = {}
+        instance["public_ips"] = [ip]
+        instance["user"] = user
+        instance["role"] = role
```
Comment (vpodzime) [inline cf_remote/commands.py:530]: I'd create the dictionary in one step instead of creating it and then populating it, but YMMV.

File: cf_remote/main.py [hunk 3]
```diff
@@ -313,7 +320,7 @@ def get_cloud_hosts(name, private_ips=False):
 
     ret = []
     for host in hosts:
-        if private_ips:
+        if private_ips and "private_ips" in host:
```
Comment (vpodzime) [inline cf_remote/main.py:323]: Why is this necessary? There's `host.get(key, [])` below so a request to get private IPs will just return empty lists for hosts that don't have private IPs. This change here twists the logic.

**Review by vpodzime — COMMENTED** [review]
Thanks for adding this.

File: cf_remote/main.py (hunk 3 above)
Comment (olehermanse) [reply cf_remote/main.py:323]: @vpodzime Yes, this changes the logic to fall back to public IPs if private IPs don't exist. This function is called with `private_ips = True` when we are looking for what IP to use for bootstrapping. In the case of saved hosts, for example from DigitalOcean, there is no private IP, so falling back to public is correct.

Comment (olehermanse) [reply cf_remote/main.py:323]: Sure, I can do that.

Comment (olehermanse) [reply cf_remote/main.py]: Added in last commit.

File: cf_remote/commands.py (hunk 2 above)
Comment (olehermanse) [reply cf_remote/commands.py]: Oki.

File: cf_remote/commands.py (hunk 1 above)
Comment (olehermanse) [reply cf_remote/commands.py]: Done.

**Review by vpodzime — APPROVED** [review]
Looks good to me now. Thanks!
